File: src/main/java/net/mreunionlabs/tapgradle/generate/GeneratePlugin.java
package net.mreunionlabs.tapgradle.generate;

import org.gradle.api.Plugin;
import org.gradle.api.Project;

/**
 * Created by abangkis on 21/02/2016.
 */
public class GeneratePlugin implements Plugin<Project> {
    @Override
    public void apply(Project project) {
        project.getExtensions().create("genExt", GeneratePluginExtension.class);
        project.getTasks().create("genStructure", GenerateTask.class);
    }
}

File: src/main/kotlin/net/mreunionlabs/tapgradle/generate/GeneratePlugin.kt
package net.mreunionlabs.tapgradle.generate

import org.gradle.api.Plugin
import org.gradle.api.Project

class GeneratePlugin : Plugin<Project> {
    override fun apply(project: Project) {
        with(project) {
            extensions.create("genExt", GeneratePluginExtension::class.java)
            val structureTask = tasks.create("genStructure", GenerateStructureTask::class.java)
            val layoutTask = tasks.create("genLayout", GenerateLayoutTask::class.java)
            layoutTask.dependsOn(structureTask)
        }
    }
}

